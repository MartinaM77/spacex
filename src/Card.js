import React from 'react';
import Moment from 'react-moment';

export default function Card(props) {
  const { mission_name, details, launch_date_local } = props;

  return (
    <div className='card'>
      <h3>{mission_name}</h3>
      <p>{details}</p>
      <p className='launchDate'>
        <Moment format='DD/MM/YYYY'>{launch_date_local}</Moment>
      </p>
    </div>
  );
}
