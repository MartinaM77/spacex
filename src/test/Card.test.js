import { render, screen } from '@testing-library/react';
import Card from '../Card';

test('render of Card matches snapshot', () => {
  const launch = {
    id: 1,
    mission_name: 'Test Mission',
    details: 'Test details of the mission',
    launch_date_local: "2021-07-16T11:42:00-04:00"
  }
  const { container } = render(<Card {...launch} />);
  expect(container.firstChild).toMatchSnapshot();
});
