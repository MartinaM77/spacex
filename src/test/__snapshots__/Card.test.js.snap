// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render of Card matches snapshot 1`] = `
<div
  class="card"
>
  <h3>
    Test Mission
  </h3>
  <p>
    Test details of the mission
  </p>
  <p
    class="launchDate"
  >
    <time
      datetime="1626450120000"
    >
      16/07/2021
    </time>
  </p>
</div>
`;
